components/eamxx/src/physics/mam/eamxx_mam_srf_and_online_emissions_process_interface.hpp

```diff
@@ -0,0 +1,133 @@
+#ifndef EAMXX_MAM_SRF_ONLINE_EMISS_HPP
+#define EAMXX_MAM_SRF_ONLINE_EMISS_HPP
+
+#include "share/grid/remap/abstract_remapper.hpp"
+#include "share/io/scorpio_input.hpp"
+
+// For MAM4 aerosol configuration
+#include <physics/mam/mam_coupling.hpp>
+#include <physics/mam/srf_emission.hpp>
+
+// For declaring surface and online emission class derived from atm process
+// class
+#include <share/atm_process/atmosphere_process.hpp>
+
+// #include <share/util/scream_common_physics_functions.hpp>
+#include <string>
+
+namespace scream {
+
+// The process responsible for handling MAM4 surface and online emissions. The
+// AD stores exactly ONE instance of this class in its list of subcomponents.
+class MAMSrfOnlineEmiss final : public scream::AtmosphereProcess {
+  using KT      = ekat::KokkosTypes<DefaultDevice>;
+  using view_1d = typename KT::template view_1d<Real>;
+  using view_2d = typename KT::template view_2d<Real>;
+
+  // number of horizontal columns and vertical levels
+  int ncol_, nlev_;
+
+  // buffer for sotring temporary variables
+  mam_coupling::Buffer buffer_;
+
+  // physics grid for column information
+  std::shared_ptr<const AbstractGrid> grid_;
+
+  // Constituent fluxes of species in [kg/m2/s]
+  view_2d constituent_fluxes_;
+
+  // Work array to store fluxes after unit conversions to kg/m2/s
+  view_1d fluxes_in_mks_units_;
+
+  // Unified atomic mass unit used for unit conversion (BAD constant)
+  static constexpr Real amufac = 1.65979e-23;  // 1.e4* kg / amu
+
+ public:
+  using srfEmissFunc = mam_coupling::srfEmissFunctions<Real, DefaultDevice>;
+
+  // Constructor
+  MAMSrfOnlineEmiss(const ekat::Comm &comm, const ekat::ParameterList &params);
+
+  // --------------------------------------------------------------------------
+  // AtmosphereProcess overrides (see share/atm_process/atmosphere_process.hpp)
+  // --------------------------------------------------------------------------
+
+  // The type of subcomponent
+  AtmosphereProcessType type() const { return AtmosphereProcessType::Physics; }
+
+  // The name of the subcomponent
+  std::string name() const { return "mam_srf_online_emissions"; }
+
+  // grid
+  void set_grids(
+      const std::shared_ptr<const GridsManager> grids_manager) override;
+
+  // management of common atm process memory
+  size_t requested_buffer_size_in_bytes() const override;
+  void init_buffers(const ATMBufferManager &buffer_manager) override;
+
+  // Initialize variables
+  void initialize_impl(const RunType run_type) override;
+
+  // Run the process by one time step
+  void run_impl(const double dt) override;
+
+  // Finalize
+  void finalize_impl(){/*Do nothing*/};
+  // Atmosphere processes often have a pre-processing step that constructs
+  // required variables from the set of fields stored in the field manager.
+  // This functor implements this step, which is called during run_impl.
+  struct Preprocess {
```

Comment:
It seems to me that this struct is pointless? Can we get rid of it?

Reply:
Yes, I have gotten rid of it.